src/preset_cli/cli/superset/sync/native/command.py

```diff
@@ -297,59 +310,115 @@ def native(  # pylint: disable=too-many-locals, too-many-arguments, too-many-bra
             configs["bundle" / relative_path] = config
 
     if split:
-        import_resources_individually(configs, client, overwrite)
+        import_resources_individually(configs, client, overwrite, continue_on_error)
     else:
         contents = {str(k): yaml.dump(v) for k, v in configs.items()}
         import_resources(contents, client, overwrite, asset_type=asset_type)
 
 
-def import_resources_individually(
+def import_resources_individually(  # pylint: disable=too-many-locals
     configs: Dict[Path, AssetConfig],
     client: SupersetClient,
     overwrite: bool,
+    continue_on_error: bool = False,
 ) -> None:
     """
     Import contents individually.
 
     This will first import all the databases, then import each dataset (together with the
     database info, since it's needed), then charts, and so on. It helps troubleshoot
     problematic exports and large imports.
+
+    By default, the import logs all assets imported correctly to a checkpoint file so that
+    if one fails, a future import continues from where it's left. If ``continue_on_error``
+    is set to True, then only failures are logged to the file, and the import continues.
     """
-    # store progress in case the import stops midway
-    checkpoint_path = Path("checkpoint.log")
-    if not checkpoint_path.exists():
-        checkpoint_path.touch()
-
-    with open(checkpoint_path, "r+", encoding="utf-8") as log:
-        imported = {Path(path.strip()) for path in log.readlines()}
-        asset_configs: Dict[Path, AssetConfig]
-        imports = [
-            ("databases", lambda config: []),
-            ("datasets", lambda config: [config["database_uuid"]]),
-            ("charts", lambda config: [config["dataset_uuid"]]),
-            ("dashboards", get_dashboard_related_uuids),
+    imports = [
+        ("databases", lambda config: []),
+        ("datasets", lambda config: [config["database_uuid"]]),
+        ("charts", lambda config: [config["dataset_uuid"]]),
+        ("dashboards", get_dashboard_related_uuids),
+    ]
+    asset_configs: Dict[Path, AssetConfig]
+    related_configs: Dict[str, Dict[Path, AssetConfig]] = {}
+
+    file_path = Path("progress.log")
+    if not file_path.exists():
+        file_path.touch()
+
+    with open(file_path, "r", encoding="utf-8") as log_file:
+        logs = yaml.load(log_file, Loader=yaml.SafeLoader) or {}
+
+    # Remove FAILED logs to re-try them
+    if "assets" in logs:
+        logs["assets"] = [
+            asset for asset in logs["assets"] if asset["status"] != "FAILED"
         ]
-        related_configs: Dict[str, Dict[Path, AssetConfig]] = {}
-        for resource_name, get_related_uuids in imports:
-            for path, config in configs.items():
-                if path.parts[1] != resource_name:
-                    continue
+    else:
+        logs["assets"] = []
+
+    assets_to_skip = {Path(log["path"]) for log in logs["assets"]} if logs else set()
```

Comment:
The `if logs else set()` is not needed, since `logs` will always have the `assets` key (set in line 354 or 358), and hence will always be truthy.

Reply:
ohh good catch! I added that enforcement after having this, and forgot to remove it -- fixing it now too.